### PR TITLE
pass parameters by value to client methods

### LINKED ISSUE #52: Pass values instead of pointers to send methods
@f2prateek 

What do you think about passing values instead of pointers to the send methods? Besides making the syntax heavier and putting more pressure on the compiler and GC I don't really see why this is done (we're not using the `nil` value for example and it would always cause a panic to pass `nil`, crashing at runtime instead of compile time).

Basically changing
```go
func (c *Client) Alias(a *Alias) error { ... }
```
to
```go
func (c *Client) Alias(a Alias) error { ... }
```
Let me know what you think about it?

### PR DESCRIPTION
**Pull request by achille-roussel**
@f2prateek 
@calvinfo 

Fixes #52 

This is an intermediary change toward #53, please review and let me know if anything should be changed.

### CODE REVIEW

**Comment by achille-roussel** [comment]
The Circle trace is strange, it looks like it tested against an old version of the library but with the newer tests:
```
godep: [WARNING]: godep should only be used inside a valid go package directory and
godep: [WARNING]: may not function correctly. You are probably outside of your $GOPATH.
godep: [WARNING]:	Current Directory: /home/ubuntu/analytics-go
godep: [WARNING]:	$GOPATH: /home/ubuntu/.go_workspace:/usr/local/go_workspace
# _/home/ubuntu/analytics-go/examples
examples/track.go:22: cannot use analytics.Track literal (type analytics.Track) as type *analytics.Track in argument to client.Track
ok  	_/home/ubuntu/analytics-go	5.013s	coverage: 62.9% of statements
godep: go exit status 2
make: *** [test] Error 1

make test returned exit code 2
```

**Comment by achille-roussel** [comment]
Definitely passes when I run it:
```
$ make test
ok  	github.com/segmentio/analytics-go	5.020s	coverage: 61.2% of statements
?   	github.com/segmentio/analytics-go/examples	[no test files]
Achilles-MacBook-Pro:analytics-go achilleroussel$ git status
On branch pass-by-value
Your branch is up-to-date with 'origin/pass-by-value'.
nothing to commit, working directory clean
```

**Comment by f2prateek** [comment]
Yup the thing with circle is that it fetches master rather than the Github src directory. We should be moving the directory into the gopath.

**Comment by achille-roussel** [comment]
@f2prateek alright it looks like the tests are passing now! I had to make changes to the tests so that they would pass `go vet`, basically I converted a bunch of examples to actual unit tests (it should have no impact since they didn't even show up properly in the doc).

Please take a look and let me know if you're good with the changes.

File: analytics_test.go [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package analytics
 
-import "net/http/httptest"
```
Comment (calvinfo) [inline analytics_test.go]: mind making these all follow the same imports? just for consistency :)

File: analytics_test.go [hunk 2]
```diff
@@ -108,44 +111,51 @@ func ExampleClose() {
...
-	//   "context": {
-	//     "library": {
-	//       "name": "analytics-go",
-	//       "version": "2.1.0"
-	//     }
-	//   },
-	//   "messageId": "I'm unique",
-	//   "sentAt": "2009-11-10T23:00:00+0000"
-	// }
+	const ref = `{
```
Comment (calvinfo) [inline analytics_test.go]: ❤️

**Comment by calvinfo** [comment]
LGTM

File: analytics_test.go (hunk 1 above)
Comment (achille-roussel) [inline analytics_test.go]: Hm... Not sure why gofmt didn't fix this, I'll do!

**Comment by achille-roussel** [comment]
Thanks for reviewing, I had based this PR on master which was a mistake, see #65 which merges those changes into `v3.0`.

Closing.

**Pull request closed without merging**